src/hotspot/cpu/x86/vm_version_x86.cpp

```diff
@@ -387,6 +421,53 @@ class VM_Version_StubGenerator: public StubCodeGenerator {
     __ movl(Address(rsi, 8), rcx);
     __ movl(Address(rsi,12), rdx);
 
+#ifndef PRODUCT
+    //
+    // Check if OS has enabled XGETBV instruction to access XCR0
+    // (OSXSAVE feature flag) and CPU supports APX
+    //
+    // To enable APX, check CPUID.EAX=7.ECX=1.EDX[21] bit for HW support
+    // and XCRO[19] bit for OS support to save/restore extended GPR state.
+    __ lea(rsi, Address(rbp, in_bytes(VM_Version::sefsl1_cpuid7_offset())));
+    __ movl(rax, 0x200000);
+    __ andl(rax, Address(rsi, 4));
+    __ cmpl(rax, 0x200000);
+    __ jcc(Assembler::notEqual, vector_save_restore);
+    // check _cpuid_info.xem_xcr0_eax.bits.apx_f
+    __ movl(rax, 0x80000);
+    __ andl(rax, Address(rbp, in_bytes(VM_Version::xem_xcr0_offset()))); // xcr0 bits apx_f
+    __ cmpl(rax, 0x80000);
+    __ jcc(Assembler::notEqual, vector_save_restore);
+
+    /* FIXME: Uncomment while integrating JDK-8329032
+    bool save_apx = UseAPX;
```

Comment:
What are you missing to uncomment this code?
8329032 is about `.ad` file changes. It should not affect execution of this code. 
You need changes in `register_x86.*` files and may be somewhere else but you don't need C2 changes for this code to work.

Reply:
Register definitions for r16 and r31.

Reply:
Yes, we already have that in place with https://github.com/openjdk/jdk/pull/19042, which will be open for review after this patch.  I added it in comments since this piece of logic is centered around CPUID feature check.

Reply:
Okay.